python/torch_mlir/extras/fx_importer.py

```diff
@@ -377,6 +610,10 @@ def import_frozen_exported_program(self, prog: torch.export.ExportedProgram):
         As we anticipate more nuanced treatment options in the future, we name this
         method to indicate that it is producing "frozen" modules. Additional top-level
         approaches to handling state can be introduced later as an addition.
+
+        TODO: This mechanism should be eventually replaced by `import_program` with
+        hooks set on the subclass to freeze parameters and buffers. However, that is
+        waiting for the Torch 2.3 release cut.
```

Comment:
I'm hoping this could be done in a backwards compatible way (i.e. to allow it to work with older Torch releases)?

Reply:
We will have to see. At best, it might be possible to retrofit to 2.2 but I'm not sure it is worth it. Further, the API is still evolving so we need to keep this separated at least until the 2.3 release so we can keep adapting it.

There are limits to what features can be supported on old versions as torch itself was missing the needed sophistication. I suspect that the narrow use case of frozen, stateless graphs is all we will manage for the older versions.